### PR TITLE
chore: update KalturaClient to v17.18.1

### PR DESCRIPTION
**Pull request by PenghaiZhang**
Due to the Log4J update, KalturaClient v3.2.1 no longer works. And this version is really too old.

This means it's time to updates KalturaClient to the latest version v17.18.1. 

However,  there are plenty of breaking changes.  So I also updated code where KalturaClient is used.

Most significant changes were made in `KalturaServiceImpl` around how to send a request to manage Kaltura resources.

I have tested:
1. Adding a new Kaltura server in the setting page. Clicking the test connection button showed me a successful connection.
2. Uploading a video in Wizard.
3. Searching for the video in Wizard.
4. Viewing the video in new search UI.

### CODE REVIEW

File: Kaltura/com.tle.core.kaltura/src/com/tle/core/kaltura/service/KalturaServiceImpl.java [hunk 1]
```diff
@@ -68,422 +70,362 @@
...
+
+  // Increase number on the end to replace
+  private static final String EQUELLA_KDP_UICONF = "EQUELLA-KDP-UICONF_5.2-114";
+
+  // A cache to protect against excessive calls to Kaltura. There is a small risk of a period when
+  // incorrect results could then be returned - however if needs be a restart can navigate it.
+  private final Cache<String, UiConf> defaultUiConfCache =
+      CacheBuilder.newBuilder().expireAfterWrite(1, TimeUnit.HOURS).build();
+
+  private <T> T execute(RequestElement<T> request) throws APIException {
```
Comment (PenghaiZhang) [inline Kaltura/com.tle.core.kaltura/src/com/tle/core/kaltura/service/KalturaServiceImpl.java:98]: Due to the Java format changes, this file is really hard to review I think. But the key change is this method and where it's used. This is the new approach KalturaClients provides to communicate with the server. We basically need to create a variety of requests, use the `APIOkRequestsExecutor` to execute them, and then check whether the request is successful or not.

Comment (PenghaiZhang) [reply Kaltura/com.tle.core.kaltura/src/com/tle/core/kaltura/service/KalturaServiceImpl.java:98]: BTW it seems a lot easier to see the differences locally with Intellij.

Comment (edalex-ian) [reply Kaltura/com.tle.core.kaltura/src/com/tle/core/kaltura/service/KalturaServiceImpl.java:98]: Thank you. I'll turn off whitespace then and see if that helps. :crossed_fingers:

**Review by edalex-ian — APPROVED** [review]
LGTM, I can see the changes are mostly naming, and then as you said, the need for that `execute` wrapper.

Big jump in dependency though, we probably want to especially call out this one in the release notes. (Please make sure you put a comment in the release notes ticket in Jira.)

**Review by ChristianMurphy — APPROVED** [review]

**Pull request merged**